src/fparser/common/tests/test_readfortran.py

```diff
@@ -340,6 +340,258 @@ def test_base_free_continuation(log):
+                cfile.write(include_info[include_filename])
+        from fparser.common.readfortran import FortranFileReader
+        reader = FortranFileReader(fortran_filename,
+                                   ignore_comments=ignore_comments)
+        for orig_line in expected.split("\n"):
+            new_line = reader.next().line
+            assert new_line == orig_line
+        with pytest.raises(StopIteration):
+            reader.next()
+    finally:
+        oldpwd.chdir()
+
+
+FORTRAN_CODE = ("program test\n"
+                "  ! prog comment 1\n"
+                "  print *, 'Hello'\n"
+                "  ! prog comment 2\n"
+                "end program")
+
+EXPECTED_CODE = ("program test\n"
+                 "print *, 'Hello'\n"
+                 "end program")
+
+
+def test_include1(tmpdir):
+    '''Test that FortranReaderBase can parse an include file when the
+    original program consists only of an include.
+
+    '''
+    fortran_filename = "prog.f90"
+    include_filename = "prog.inc"
+    fortran_code = ("include '{0}'".format(include_filename))
+    include_code = EXPECTED_CODE
+    include_info = {}
+    include_info[include_filename] = include_code
+    check_include_works(fortran_filename, fortran_code, include_info,
+                        EXPECTED_CODE, tmpdir)
+
+
+def test_include2(tmpdir):
+    '''Test that FortranReaderBase can parse an include file when the
+    original and include files both files have multiple lines.
+
+    '''
+    fortran_filename = "prog.f90"
+    include_filename = "my-include.h"
+    fortran_code = ("module include_test\n"
+                    "  include '{0}'\n"
+                    "end module include_test".format(include_filename))
+    include_code = ("interface mpi_sizeof\n"
+                    "subroutine simple()\n"
+                    "end subroutine simple\n"
+                    "end interface mpi_sizeof")
+    split_code = fortran_code.split("\n")
+    expected = split_code[0] + "\n" + include_code + "\n" + split_code[2]
+    include_info = {}
+    include_info[include_filename] = include_code
+    check_include_works(fortran_filename, fortran_code, include_info,
+                        expected, tmpdir)
+
+
+def test_include3(tmpdir):
+    '''Test that FortranReaderBase can parse an include file when the
+    original program is invalid without the include.
+
+    '''
+    fortran_filename = "prog.f90"
+    include_filename = "prog.inc"
+    fortran_code = ("program test\n"
+                    "include '{0}'".format(include_filename))
+    include_code = ("print *, 'Hello'\n"
+                    "end program")
+    include_info = {}
+    include_info[include_filename] = include_code
+    check_include_works(fortran_filename, fortran_code, include_info,
+                        EXPECTED_CODE, tmpdir)
+
+
+def test_include4(tmpdir):
+    '''Test that FortranReaderBase can parse a multiple include files.'''
```

Comment:
s/a multiple/a source file containing multiple/

Reply:
Done, but slightly modified you suggestion as ReaderBase mitgh not read from a file.